history.go

```diff
@@ -54,11 +68,15 @@ func (h *History) UpdateHistory(policies Policies, m *MerkleTree, nextEp Epoch)
 		}
 		h.loadedEpochs = append(h.loadedEpochs[:0], h.loadedEpochs[n:]...)
 	}
-	h.snapshots[nextEp] = nextStr
-	h.loadedEpochs = append(h.loadedEpochs, nextEp)
+
+	h.updateInternal(policies, nextEp)
 	return nil
 }
 
+func (h *History) Set(key string, value []byte) error {
```

Comment:
On my way out, but I wanted to leave a pointer to this comment, in case it's relevant here.  https://github.com/coniks-sys/libmerkleprefixtree-go/pull/8#issuecomment-231272272